### PR TITLE
fix(components): [el-table] width init error (close #3087)

### LINKED ISSUE #3087: [Bug Report] el-space incrementally growing horizontally by itself without end
### Element Plus version
1.1.0-beta.6

### OS/Browsers version
Mac OS Big Sur - Firefox/Safari

### Vue version
3.2.6

### Reproduction Link
https://jsfiddle.net/1bd4sLto/

### Steps to reproduce
Simply placing an el-table inside an el-space with :fill=true will cause the div to expand forever

            <el-space :fill="true">
                <el-table>
                    <el-table-column>This will be expanding horizontally</el-table-column>
                </el-table>
            </el-space>

### What is Expected?
Expected for the el-space to take the max available width and stop moving

### What is actually happening?
From t=0 of render time, the width of <main class="el-main"> element keeps increasing forever

### PR DESCRIPTION
**Pull request by imswk**
fix(#3087)

* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by JeremyWuuuuu** [comment]
May I ask why do we remove `resize` event listener here?

**Comment by imswk** [comment]
> May I ask why do we remove `resize` event listener here?

#3087 

`el-space` style` min-width: 100%;` It will affect the value of `resize`. I don't think it's good to delete it directly. I've modified it.

File: packages/components/table/src/table/style-helper.ts [hunk 1]
```diff
@@ -153,22 +153,24 @@ function useStyle<T>(
...
 
   const bindEvents = () => {
-    window.addEventListener('resize', doLayout)
     table.refs.bodyWrapper.addEventListener('scroll', syncPostion, {
       passive: true,
     })
     if (props.fit) {
       addResizeListener(table.vnode.el as ResizableElement, resizeListener)
+    } else {
+      window.addEventListener('resize', doLayout)
```
Comment (JeremyWuuuuu) [inline packages/components/table/src/table/style-helper.ts:162]: Use `on` and `off` exported from `@element-plus/utils/dom`

Comment (imswk) [reply packages/components/table/src/table/style-helper.ts]: OK，I'm done 🙇

**Review by sxzz — APPROVED** [review]

**Pull request merged**